File: src/Issue.Web/ViewModels/AccountModels.cs
using System.ComponentModel.DataAnnotations;

namespace Issue.Web.ViewModels
{
    public class LoginViewModel
    {
        [Required]
        [MaxLength(200)]
        public string Username { get; set; }

        [Required]
        [MaxLength(200)]
        public string Password { get; set; }
    }

    public class LoginResult
    {
        public bool Success { get; set; }
        public string[] Roles { get; set; }
        public int UserId { get; set; }
    }

    public class UserViewModel
    {
        public int Id { get; set; }
        
        [Required]
        public string Email { get; set; }
        [Required]
        [MinLength(6)]
        public string Password { get; set; }

        public string PhoneNumber { get; set; }

        public string UserName { get; set; }

    }

    public class UserUpdateModel
    {
        public int Id { get; set; }

        [Required]
        public string Email { get; set; }
      
        public string PhoneNumber { get; set; }
        [Required]
        public string UserName { get; set; }
    }

    public class UserForListModel
    {
        public int Id { get; set; }
        public string Email { get; set; }

        public string UserName { get; set; }
    }

    public class UpdateUserPassword
    {
        [Required]
        public int Id { get; set; }

        [Required]
        [StringLength(20, MinimumLength = 6)]
        public string Password { get; set; }

        [Compare("Password")]
        public string ConfirmPassword { get; set; }
    }
}

File: src/Issue.Web/ClientApp/src/app/models/AccountModels.ts
export class LoginViewModel {
  username: string;
  password: string;
}

export class TokenViewModel {
  access_token: string;
  expires_in: Date;
  refresh_token: string;
  roles: string[];
}

export class UserForListModel {
  id: number;
  email: string;
  userName: string;
}

export class UserViewModel {
  id: number;
  email: string;
  password: string;
  phoneNumber: string;
  userName: string;
}

export class UserUpdateModel {
  id: number;
  email: string;
  phoneNumber: string;
  userName: string;
}



export class UpdateUserPassword {
  id: number;
  password: string;
  confirmPassword: string;
}
